test/api_compatibility.js

```diff
@@ -103,30 +96,78 @@ module.exports = function apiCompatibilityChecks(_createTracer) {
             });
         });
 
-        describe('SpanContext', function() {
-            var tracer = createTracer();
-            var span = tracer.startSpan('test_span');
-            var spanContext = span.context();
+        describe('Reference', function() {
+            it('should handle Spans and SpanContexts', function() {
+                expect(function() { new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, span); }).to.not.throw(Error);
+            });
+        });
+    });
 
-            it('should be a class', function() {
-                expect(spanContext).to.be.an('object');
+    describe('Opentracing implementation compatibility', function() {
+        var tracer;
+        var span;
+
+        beforeEach(function() {
+            tracer = createTracer();
+            span = tracer.startSpan('test-span');
+        });
+
+        describe('Tracer', function() {
+            _.each(spanFunctions, function(name) {
+                it(name + ' should be a method', function() {
+                    var span = tracer.startSpan('test-span');
+                    expect(span[name]).to.be.a('function');
+                });
+            });
+
+            it ('extract returns span context with allowed functions', function() {
+                // pass by default if no factory map was specified.
+                if (carrierFormatFactoryMap) {
+                    var carrierFactory = carrierFormatFactoryMap[opentracing.FORMAT_TEXT_MAP];
+                    tracer.extract(opentracing.FORMAT_TEXT_MAP, carrierFactory());
```

Comment:
the description does not match test behavior - there are no assertions in the test